lib/enkf/local_dataset.cpp

```diff
@@ -24,14 +24,15 @@
 
 #include <ert/enkf/enkf_macros.hpp>
 #include <ert/enkf/local_ministep.hpp>
-
+#include <ert/enkf/row_scaling.hpp>
 
 #define LOCAL_DATASET_TYPE_ID  6615409
 
 struct local_dataset_struct {
   UTIL_TYPE_ID_DECLARATION;
   char      * name;
-  hash_type * nodes;            /* A hash table indexed by node keys - each element is an active_list instance. */
+  hash_type * active_size;      /* A hash table indexed by node keys - each element is an active_list instance. */
```

Comment:
Is this better than `std::unordered_map`?

Reply:
`std::unordered_map` is probably better; a it more hassle to wrap the pointer in something which will destruct it.

Reply:
It's kind of ugly, but:
```c++
struct active_list_deleter {
    void operator()(active_list *p) const {
        active_list_free(p);
    }
};

/* Not sure about the naming convention */
using active_list_uq = std::unique_ptr<active_list, active_list_deleter>;

struct local_dataset_struct {
    /* [..] */
    std::unordered_map<std::string, active_list_uq> active_size;
}
```

It's easier if the objects are C++ classes that are wrapped by some C classes, however.

Reply:
Yes - I know; that is the alternative - in my opinion this is a bit halfways; in my opinion the alternatives are:

1. Full C style approach - for all it is worth that pattern is used extensively in the code.
2. Full C++ style with `std::unordered_map<ActiveList>` and a full blown `class ActiveList` with value semantics.

I'll do as told though :-)